pkg/datapath/garp/garp.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package garp
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/mdlayher/arp"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+const (
+	// GARPInterface is the interface used to send Gratuitous ARP messages.
+	GARPInterface = "garp-interface"
+)
+
+var Cell = cell.Module(
+	"garp",
+	"GARP",
+
+	cell.Provide(newGARPSender),
+
+	// This cell can't have a default config, it's entirely env dependent.
+	cell.Config(Config{}),
+)
+
+// Config contains the configuration for the GARP cell.
+type Config struct {
+	GARPInterface string
+}
+
+func (def Config) Flags(flags *pflag.FlagSet) {
+	flags.String(GARPInterface, def.GARPInterface, "Interface used for sending gratuitous arp messages")
+}
+
+type Sender interface {
+	Send(netip.Addr) error
+}
+
+func newGARPSender(log logrus.FieldLogger, cfg Config) (Sender, error) {
+	if cfg.GARPInterface == "" {
+		return nil, errors.New("gratuitous arp sender interface undefined")
+	}
+
+	iface, err := net.InterfaceByName(cfg.GARPInterface)
+	if err != nil {
+		return nil, fmt.Errorf("gratuitous arp sender interface %q not found: %w", cfg.GARPInterface, err)
+	}
+
+	l := log.WithField("interface", iface.Name)
+	l.Info("initialised gratuitous arp sender")
+
+	return &sender{
+		logger: l,
+		iface:  iface,
+	}, nil
+}
+
+type sender struct {
+	logger logrus.FieldLogger
+
+	iface *net.Interface
+}
+
+// Send implements Sender
+func (s *sender) Send(ip netip.Addr) error {
+	arpClient, err := arp.Dial(s.iface)
```

Comment:
Do we need to create the client each time or can it be initialized once in `newGARPSender`?

Reply:
The client that's created uses a raw socket underneath. I'm worried about keeping a raw socket open all the time, and figured it would be better to create and destroy on-demand. If there's no issue with keeping raw sockets open, then I'm happy to create the client in the constructor.